File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        
        String str = x + "";
        int i=0;
        int j=str.length()-1;
        
        while(i<j) {
            if(str.charAt(i++) != str.charAt(j--)) return false;
        }
        
        return true;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        
        if(x<0) return false;
        
        vector<int> a;
        while(x>0) {
            a.push_back(x%10);
            x /= 10;
        }
        
        for(int i=0;i<a.size();i++) {
            if(a[i] != a[a.size()-i-1]) return false;
        }
        
        return true;
    }
};